### PR TITLE
User story 6: Add a book

### LINKED ISSUE #6: [US-06] Add new book to list of books currently read
As a user I want to navigate to a page where I can add a new book to my list of books currently read.

## Description
addbooktonav

## Acceptance criteria
- [x] User can enter information about the book (title and author) and reading status (since when the book is read and on which page the user is currently) and also upload an image
- [x] Title and author are required for adding a new book
- [x] If there is no image a placeholder image will be added
- [x] After completing the form the user should be redirected to list of books currently read and see the book on top of the booklist
- [x] Navigation should show three navlinks("Currently reading", "Library" & "Add new book"
- [x] The user should be directed to the page where he can add a new book to the list of books currently read by clicking on the navlink "Add new book"

## Tasks

- [x] Create feature branch 
- [x] Create add book component
- [x] Change component for list of books currently read
- [x] Write tests for new component
- [x] Add component to App
- [x] Change navigation component
- [x] Change test for components if necessary
- [x] Add route for component "AddBook" in App
- [x] Add and adjust styled components

### PR DESCRIPTION
**Pull request by mbosselmann**
This resolves #6 

- Implement new AddBook component (with story and tests)
- Add AddBook to App (routing and props)
- Move general form styles from Start to GlobalStyles to use these styles again in AddBook component
- Add new NavLink in Navigation for AddBook

Feedback is much appreciated. :-)

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+import styled from 'styled-components/macro'
+import { nanoid } from 'nanoid'
+import placeholder from '../images/placeholder.png'
+import previewPlaceholder from '../images/preview-placeholder.png'
+
+function AddBook({ books, setBooks }) {
+  const [bookcover, setBookcover] = useState(placeholder)
+  const history = useHistory()
+  function getBookcoverPreview(previewEvent) {
+    const preview = URL.createObjectURL(previewEvent.target.files[0])
```
Comment (StefanBehring) [inline src/components/AddBook.js:12]: Out of curiosity: What does this do exactly?

**Review by StefanBehring — COMMENTED** [review]
Really well done again! Added some suggestions/questions.

File: src/components/AddBook.stories.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import AddBook from './AddBook'
+
+export default {
+  title: 'Component/AddBook',
+  component: AddBook,
+}
+
+const Template = args => <AddBook {...args} />
```
[the remark this answers is not in the document]
Comment (mbosselmann) [reply src/components/AddBook.stories.js:8]: Thank you! I didn't know yet! Will change this for my components. :-)

File: src/components/AddBook.js (hunk 1 above)
Comment (mbosselmann) [reply src/components/AddBook.js:12]: I want to show a preview image after uploading the bookcover file in my addbook page and with URL.createObjectURL I can make a URL for this file and display it. :-) Read about it here: [URL.createObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)

**Pull request merged**